enterprise/server/crypter/crypter.go

```diff
@@ -0,0 +1,255 @@
+package crypter
+
+import (
+	"context"
+	"crypto/cipher"
+	"crypto/rand"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/metrics"
+	"github.com/buildbuddy-io/buildbuddy/server/remote_cache/digest"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"golang.org/x/crypto/chacha20poly1305"
+
+	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+	sgpb "github.com/buildbuddy-io/buildbuddy/proto/storage"
+)
+
+const (
+	// This must be exposed because it is used as an input to the
+	// key-derivation function in crypter_service.
+	EncryptedDataHeaderVersion = 1
+
+	encryptedDataHeaderSignature = "BB"
+	nonceSize                    = chacha20poly1305.NonceSizeX
+	encryptedChunkOverhead       = nonceSize + chacha20poly1305.Overhead
+)
+
+// An encryption key. Note that this is the "derived" key as described in
+// http://go/customer-managed-encryption.
+type Key struct {
```

Comment:
Let's call this `DerivedKey`? Information is more useful in the name than in in a comment (not to say that comments are not useful).